src/Microsoft.DotNet.Build.Tasks/CreateAzureContainer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class CreateAzureContainer : Task
+    {
+        [Required]
+        public string AccountName { get; set; }
+
+        [Required]
+        public string AccountKey { get; set; }
+
+        [Required]
+        public string ContainerName { get; set; }
+
+        [Output]
+        public string ReadOnlyToken { get; set; }
+
+        [Output]
+        public string StorageUri { get; set; }
+
+        [Output]
+        public string WriteOnlyToken { get; set; }
+
+        public override bool Execute()
+        {
+            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
+
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
+            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
+            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            newContainer.CreateIfNotExists();
+            StorageUri = newContainer.Uri.ToString();
+
+            // if ReadOnlyToken or WriteOnlyToken != null they should contain the number of days before the token expires
+
+            if (ReadOnlyToken != null)
```

Comment:
I was trying to reduce the number of properties.  How this works is when calling CreateAzureContainer if a RO token is required you specify the number of days for which it should last in ReadOnlyToken.  The token is created and then placed in ReadOnlyToken.  I suppose the complexity is the same either way (i.e. lots of properties or this behavior) so if people find it confusing I will change it to explicit properties.